cpu/saml21/cpu.c

```diff
@@ -64,6 +64,53 @@ static void _xosc32k_setup(void)
 #endif
 }
 
+static void _dfll_setup(void)
+{
+#if (CLOCK_CORECLOCK == 48000000U) || defined (MODULE_PERIPH_USBDEV)
```

Comment:
Why not put the function definition inside the macros:
```c
#if (CLOCK_CORECLOCK == 48000000U) || defined (MODULE_PERIPH_USBDEV)
static void _dfll_setup(void)
{
...
}
#endif
```

and later call the function only when core clock is 48MHz and usbdev module is used.

Reply:
This would force me to add another guard when calling _dfll_setup and thus decrease readability. Should I change it anyways ?

Reply:
> add another guard when calling _dfll_setup

I know ;)

> Should I change it anyways ?

It depends: if it saves some ROM when using a 16MHz coreclock then yes.

Reply:
> It depends: if it saves some ROM when using a 16MHz coreclock then yes.

I tested this specific point, there is no RAM/ROM changes if we keep it this way. Hopefully :)